scripts/mkstatic.py

```diff
@@ -1263,6 +1318,145 @@ def write(self, metadata_request, metadata_define, arguments, debug):
+                                    # Handle horizontal dimensions correctly
+                                    if ccpp_stage == 'run':
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        if dims[1] == CCPP_HORIZONTAL_LOOP_EXTENT and not dim0:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        elif CCPP_HORIZONTAL_LOOP_BEGIN in dims or CCPP_HORIZONTAL_LOOP_END in dims or \
+                                                CCPP_HORIZONTAL_DIMENSION in dims:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                    else:
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        if dims[1] == CCPP_HORIZONTAL_DIMENSION and not dim0:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                        # This should not happen when parsing metadata with capgen's metadata parser, remove?
+                                        if CCPP_HORIZONTAL_LOOP_BEGIN in dims or CCPP_HORIZONTAL_LOOP_END in dims or \
+                                                CCPP_LOOP_EXTENT in dims:
+                                            raise Exception(f"Invalid metadata for scheme {scheme_name}: " + \
+                                                            f"horizontal dimension for {var_standard_name} is {var.dimensions}")
+                                    # *DH
+
+                                # DH* TODO: move this to a function "is_horizontal_dimension" ?
+                                # maybe combine with some of the above logic to get the correct dimensions
+                                # depending on phase etc?
+                                # We only want to be explicit about horizontal dimensions, for which we need
+                                # to subset in certain cases (therefore do it always). Other dimensions are
+                                # passed as ':'.
+                                #
+                                # DH* TODO: WE CANNOT ACTIVATE USING EXPLICIT HORIZONTAL BLOCKS
```

Comment:
Same as above, we need to fix the way we handle inactive variables to make the switch to contiguous arrays